server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesCreationProcessor.java

```diff
@@ -134,6 +135,9 @@ private void createMailbox(MailboxCreationId mailboxCreationId, MailboxCreateReq
 
     private void ensureValidMailboxName(MailboxCreateRequest mailboxRequest, MailboxSession mailboxSession) {
         String name = mailboxRequest.getName();
+        if (name.length() >= MailboxConstants.DEFAULT_LIMIT_MAILBOX_NAME_SIZE) {
+            throw new MailboxNameException(String.format("The mailbox '%s' is over limitation: %s", name, MailboxConstants.DEFAULT_LIMIT_MAILBOX_NAME_SIZE));
```

Comment:
The mailbox name length is over limitation

is even more clear as an error message, no ?